src/main/java/graphql/schema/impl/SchemaUtil.java

```diff
@@ -100,6 +100,7 @@ public static void replaceTypeReferences(GraphQLSchema schema) {
         final Map<String, GraphQLNamedType> typeMap = schema.getTypeMap();
         List<GraphQLSchemaElement> roots = new ArrayList<>(typeMap.values());
         roots.addAll(schema.getDirectives());
+        roots.addAll(schema.getSchemaAppliedDirectives());
```

Comment:
This makes sense to me @andimarek - these objects are not in the schema (eg linked by other objects) - they are root level objects